content/en/docs/tasks/administer-cluster/kubeadm/kubeadm-upgrade-1-13.md

```diff
@@ -0,0 +1,278 @@
+
+    On the master node, you must add `--ignore-daemonsets`:
+
+    ```shell
+    kubectl drain ip-172-31-85-18
+    node "ip-172-31-85-18" cordoned
+    error: unable to drain node "ip-172-31-85-18", aborting command...
+
+    There are pending nodes to be drained:
+    ip-172-31-85-18
+    error: DaemonSet-managed pods (use --ignore-daemonsets to ignore): calico-node-5798d, kube-proxy-thjp9
+    ```
+
+    ```
+    kubectl drain ip-172-31-85-18 --ignore-daemonsets
+    node "ip-172-31-85-18" already cordoned
+    WARNING: Ignoring DaemonSet-managed pods: calico-node-5798d, kube-proxy-thjp9
+    node "ip-172-31-85-18" drained
+    ```
+
+1.  Upgrade the Kubernetes package version on each `$NODE` node by running the Linux package manager for your distribution:
+
+    {{< tabs name="k8s_install" >}}
+    {{% tab name="Ubuntu, Debian or HypriotOS" %}}
+    apt-get update
+    apt-get upgrade -y kubelet kubeadm
+    {{% /tab %}}
+    {{% tab name="CentOS, RHEL or Fedora" %}}
+    yum upgrade -y kubelet kubeadm --disableexcludes=kubernetes
+    {{% /tab %}}
+    {{< /tabs >}}
+
+## Upgrade kubelet on each node
+
+1.  On each node except the master node, upgrade the kubelet config:
+
+    ```shell
+    kubeadm upgrade node config --kubelet-version $(kubelet --version | cut -d ' ' -f 2)
+    ```
+
+1.  Restart the kubelet process:
+
+    ```shell
+    systemctl restart kubelet
+    ```
+
+1.  Verify that the new version of the `kubelet` is running on the node:
+
+    ```shell
+    systemctl status kubelet
+    ```
+
+1.  Bring the node back online by marking it schedulable:
+
+    ```shell
+    kubectl uncordon $NODE
+    ```
+
+1.  After the kubelet is upgraded on all nodes, verify that all nodes are available again by running the following command from anywhere kubectl can access the cluster:
+
+    ```shell
+    kubectl get nodes
+    ```
+
+    The `STATUS` column should show `Ready` for all your nodes, and the version number should be updated.
+
+{{% /capture %}}
+
+## Recovering from a failure state
+
+If `kubeadm upgrade` fails and does not roll back, for example because of an unexpected shutdown during execution, you can run `kubeadm upgrade` again.
+This command is idempotent and eventually makes sure that the actual state is the desired state you declare.
+
+To recover from a bad state, you can also run `kubeadm upgrade --force` without changing the version that your cluster is running.
+
+## How it works
+
+`kubeadm upgrade apply` does the following:
+
+- Checks that your cluster is in an upgradeable state:
+  - The API server is reachable
+  - All nodes are in the `Ready` state
+  - The control plane is healthy
+- Enforces the version skew policies.
+- Makes sure the control plane images are available or available to pull to the machine.
+- Upgrades the control plane components or rollbacks if any of them fails to come up.
+- Applies the new `kube-dns` and `kube-proxy` manifests and enforces that all necessary RBAC rules are created.
```

Comment:
We can say "enforces that all necessary RBAC rules" or "ensures that all necessary RBAC rules are created", right?